packages/stock-location/src/repositories/stock-location.ts

```diff
@@ -0,0 +1,108 @@
+import { Context, CreateStockLocationInput, FindOptions } from "@medusajs/types"
+import { StockLocation, StockLocationAddress } from "../models"
+
+import { AbstractBaseRepository } from "./base"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { InjectTransactionManager, MedusaContext } from "@medusajs/utils"
+import {
+  FilterQuery as MikroQuery,
+  LoadStrategy,
+  FindOptions as MikroOptions,
+  wrap,
+} from "@mikro-orm/core"
+
+type InjectedDependencies = { manager: SqlEntityManager }
+
+// eslint-disable-next-line max-len
+export class StockLocationRepostiory extends AbstractBaseRepository<StockLocation> {
+  protected readonly manager_: SqlEntityManager
+
+  constructor({ manager }: InjectedDependencies) {
+    // @ts-ignore
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments)
+    this.manager_ = manager
+  }
+
+  async find(
+    options?: FindOptions<StockLocation> | undefined,
+    context: Context = {}
+  ): Promise<StockLocation[]> {
+    const [items] = await this.findAndCount(options, context)
+    return items
+  }
+
+  async findAndCount(
+    options?: FindOptions<StockLocation> | undefined,
+    context: Context = {}
+  ): Promise<[StockLocation[], number]> {
+    let manager = (context.transactionManager ??
+      this.manager_) as SqlEntityManager
+    // TODO: update it with the injectManager when its merged
+    manager = manager.fork()
```

Comment:
I've merged the pr, you can now include @InjectManager. Take a look here - https://github.com/medusajs/medusa/pull/4535/files#diff-7349cef426f2eb03140056d1c5e6dcc0714bd3967601dbaaf77c2c404891c54bR35-R40